### ISSUE TITLE
Dynamic versioning plugin not supported on Anaconda

### ISSUE THREAD

**Issue by Zeitsperre** [metalink-dev/pymetalink#28]
FYI @cehbrecht 

It seems that the versioning plugin I opted for (`poetry-version-plugin`) is not supported on conda-forge. This means that we can't set the version string for `pymetalink` when using the conda package. I had very little background in `poetry` (but more now) when I made that call. My bad!

A similar project that handles git-based tags and is supported on `conda-forge` is [poetry-dynamic-versioning](https://pypi.org/project/poetry-dynamic-versioning/). Replacing this would be very simple.

Will open a PR shortly.

**Comment by bneijt** [comment]
I think I would rather just go without any plugin. I'm perfectly fine with having just a poetry version command that copies the tag in there and builds a package based on the tag.

Dynamic versioning plugins are very handy if you release development builds and have CD where you publish offsets from a prod release to dev/test environments. We are not doing any of that here.

Do you have any information on what is wrong on conda-forge for the versioning? I suspect it might also not help that we released versions without setting a patch number (no semver triplet), so I decided to release a semver triplet 6.5.1 as well. Do you get the same issue with that release and could you add some information or references to what is going wrong?

**Comment by Zeitsperre** [comment]
> Do you have any information on what is wrong on conda-forge for the versioning? 

The issue is that I implemented a Poetry plugin that wasn't very mature (v0.x) and failed to see that there were better options out there. My packaging experience is with `setuptools`, `flit`, and `pybind11`, so I made a rookie call.

> I suspect it might also not help that we released versions without setting a patch number (no semver triplet), so I decided to release a semver triplet 6.5.1 as well. 

This is a good idea. I mentioned that it might be good to use a consistent versioning scheme for the tags as well. Couldn't hurt.

> Do you get the same issue with that release and could you add some information or references to what is going wrong?

Put simply, there are many ways to bake the version information into libraries. My impression was that the plugin I chose was able to set that properly for libraries installed via Poetry/pip, but when trying to gather the version string on conda-forge, it simply wasn't able to find it (reported version `0.0.0`). In some cases (like `setuptools_scm`), one simply needs to add that plugin to their conda-forge recipe, but given that this plugin was very much in alpha, it wasn't available on conda-forge (and I wasn't going to bother adding it).

Sorry again for the mess.
